persistence-api/src/test/java/io/stargate/db/limiter/AsyncRateLimiterTest.java

```diff
@@ -0,0 +1,150 @@
+package io.stargate.db.limiter;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import com.google.common.util.concurrent.Uninterruptibles;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicLong;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Note: The test is made to pass when the infrastructure does not give it the right timings. This means that any
+ * failure, however rare or flaky, must be taken seriously as it signifies a real problem with the code.
+ */
+public class AsyncRateLimiterTest
```

Comment:
The code fomatter does not like this file :shrug:

Reply:
My bad, forget to run it. Push-fixed.